### PR TITLE
Fix \yii\data\Sort

### PR DESCRIPTION
**Pull request by lav45**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | #13260

Tested this functionality on a production project

### CODE REVIEW

**Comment by samdark** [comment]
What issue is it fixing? The one referenced was taken care of by @klimov-paul already, wasn't it?

File: framework/data/Sort.php [hunk 1]
```diff
@@ -229,7 +229,7 @@ public function getOrders($recalculate = false)
                     $orders[$name] = $dir;
                 }
             } else {
-                $orders[] = $columns;
+                $orders = $columns;
```
Comment (cebe) [inline framework/data/Sort.php]: in this case `columns` is not an array. The return type of this method is `array`, so assigning columns to orders does not make sense.

**Review by cebe — CHANGES_REQUESTED** [review]

File: framework/data/Sort.php (hunk 1 above)
Comment (LAV45) [inline framework/data/Sort.php]: @cebe, updated

Comment (cebe) [inline framework/data/Sort.php]: changing the return type may break code that relies on current behavior. SqlDataProvider does expect an array for example: https://github.com/yiisoft/yii2/blob/4081a4b20349974ea9f37957b166fa4398cf3a8b/framework/data/SqlDataProvider.php#L121-L126

please elaborate on why you think this change should be made. It looks wrong to me.

Comment (samdark) [inline framework/data/Sort.php]: Same here. I don't see the reason for the change...

**Pull request closed without merging**